atlasdb-commons/src/main/java/com/palantir/util/AggregatingVersionedSupplier.java

```diff
@@ -48,6 +49,14 @@ public AggregatingVersionedSupplier(Function<Collection<T>, T> aggregator, long
                 .memoizeWithExpiration(this::recalculate, expirationMillis, TimeUnit.MILLISECONDS);
     }
 
+    public static <C extends Comparable<C>> AggregatingVersionedSupplier<C> min(long expirationMillis) {
+        return new AggregatingVersionedSupplier<>(AggregatingVersionedSupplier::minimum, expirationMillis);
+    }
+
+    private static <C extends Comparable<C>> C minimum(Collection<C> currentValues) {
+        return currentValues.stream().min(Comparator.naturalOrder()).orElse(null);
+    }
```

Comment:
nit: probably just pick one of `min` or `minimum`?

We can implement this in log-time by maintaining a sorted multi-set, but probably fine as is :)